src/ctx.rs

```diff
@@ -67,16 +67,17 @@ impl Ctx {
         Ok(ctx)
     }
 
-    pub async fn start(&mut self, mut rx: Receiver<CtxCommand>) {
+    pub async fn start(&mut self, mut rx: Receiver<CtxCommand>) -> anyhow::Result<()> {
         while let Some(msg) = rx.recv().await {
             match msg {
                 CtxCommand::AddTransaction(tx) => self.handle_tx(tx),
                 CtxCommand::GenerateNewBlock => self.new_block(),
                 CtxCommand::SaveOnDisk => {
-                    let _ = self.save_on_disk();
+                    self.save_on_disk()?;
```

Comment:
an error here should stop message consumption ?

Reply:
right, merged too quickly

Reply:
fixed in 05991b7